contracts/staking/StakingMessages.sol

```diff
@@ -233,22 +290,26 @@ library StakingMessages {
         uint64 nonce,
         uint64 weight
     ) internal pure returns (bytes memory) {
-        bytes memory res = new bytes(52);
+        bytes memory res = new bytes(54);
+        // Pack the codec ID.
+        for (uint256 i; i < 2; ++i) {
+            res[i] = bytes1(uint8(CODEC_ID >> (8 * (1 - i))));
+        }
         // Pack the type ID.
         for (uint256 i; i < 4; ++i) {
-            res[i] = bytes1(uint8(SET_SUBNET_VALIDATOR_WEIGHT_MESSAGE_TYPE_ID >> (8 * (3 - i))));
+            res[i + 2] = bytes1(uint8(SET_SUBNET_VALIDATOR_WEIGHT_MESSAGE_TYPE_ID >> (8 * (3 - i))));
```

Comment:
Manually hardcoding the starting offset in each for loops seems error prone. Is storing an extra offset variable that is used across loops expensive from a gas perspective? Something like "res[offset++] = ... "

Reply:
Yes, all of the packing/unpacking code is a prime candidate for optimization or replacement with something more airtight (see for example https://github.com/ava-labs/teleporter/pull/482). I'm going to consider that out of scope for this PR though, since it adheres to the existing pattern.